airbyte-integrations/connectors/source-hubspot/source_hubspot/api.py

```diff
@@ -699,6 +699,76 @@ class FormStream(Stream):
     created_at_field = "createdAt"
 
 
+class PropertyHistoryStream(IncrementalStream):
+    """Contacts Endpoint, API v1
+    Is used to get all Contacts and the history of their respective
+    Properties. Whenever a property is changed it is added here.
+    Docs: https://legacydocs.hubspot.com/docs/methods/contacts/get_contacts
+    """
+    more_key = "has-more"
+    url = "/contacts/v1/lists/all/contacts/all"
+    updated_at_field = "timestamp"
+    created_at_field = "timestamp"
+    data_field = "contacts"
+    page_field = "vid-offset"
+    page_filter = "vidOffset"
+
+    def list(self, fields) -> Iterable:
+        properties = self._api.get(f"/properties/v2/contact/properties")
+        properties_list = [single_property["name"] for single_property in properties]
+        params = {"propertyMode": "value_and_history", "property": properties_list}
+        yield from self.read(partial(self._api.get, url=self.url), params)
+
+    def _transform(self, records: Iterable) -> Iterable:
+        record: Dict[str, Dict]
+        for record in records:
+            properties = record.get("properties")
+            vid = record.get("vid")
+            value_dict: Dict
+            for key, value_dict in properties.items():
+                versions = value_dict.get("versions")
+                if key == "lastmodifieddate":
+                    continue
+                if versions:
+                    for version in versions:
+                        version["timestamp"] = self._field_to_datetime(version["timestamp"]).to_datetime_string()
+                        version["property"] = key
+                        version["vid"] = vid
+                        yield version
+
+
+class FormSubmssionStream(Stream):
+    """Submissions for forms, v1
+    Get the Submissions for a form and the field values.
+    Docs: https://legacydocs.hubspot.com/docs/methods/forms/get-submissions-for-a-form
+    """
+    url = "/form-integrations/v1/submissions/forms"
```

Comment:
Same question as [here](https://github.com/airbytehq/airbyte/pull/7787/files#r765321373)

Reply:
This stream has merely two real columns which hold the most information after it's transformed. One for a name of a form field and one for the value which was submitted. Otherwise we would have a huge table with n columns where n is the number of different form fields accross all forms. Also with every new form the schema would need to be updated. 
This goes also for the contact PropertyHistory